src/process_wrap.cc

```diff
@@ -121,8 +123,10 @@ class ProcessWrap : public HandleWrap {
     // options.uid
     Local<Value> uid_v = js_options->Get(env->uid_string());
     if (uid_v->IsInt32()) {
-      int32_t uid = uid_v->Int32Value();
-      if (uid & ~((uv_uid_t) ~0)) {
+      int32_t uid_i32 = uid_v->Int32Value();
+      uv_uid_t uid = static_cast<uv_uid_t>(uid_i32);
+      // uv_uid_t may be unsigned, so compare with the widest available type
+      if (static_cast<intmax_t>(uid_i32) != static_cast<intmax_t>(uid)) {
```

Comment:
> People sometimes pass e.g. `{ uid: -2 }` but this check would reject that.

This check would reject that iff `uv_uid_t` is unsigned, which is kind of the point here… or am I missing something? 
And yes, that the conversion is narrowing is intentional – the goal is to see whether the value is still the “same” after converting to `uv_uid_t`.